app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyNpsQuestionScreen.kt

```diff
@@ -29,59 +28,53 @@ import com.kks.nimblesurveyjetpackcompose.model.SurveyAnswer
 import com.kks.nimblesurveyjetpackcompose.ui.theme.White50
 
 private const val START_INDEX = 0
+private const val MAX_NUMBER_OF_NPS_QUESTION = 10
 
-@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun SurveyNpsQuestionScreen(
-    answers: List<SurveyAnswer>,
-    onChooseAnswer: (answers: List<SurveyAnswer>) -> Unit
+    answers: List<SurveyAnswer>, onChooseAnswer: (answers: List<SurveyAnswer>) -> Unit
 ) {
     var selectedIndex by remember { mutableStateOf(answers.indexOfFirst { it.selected }) }
 
     ConstraintLayout(modifier = Modifier.wrapContentWidth()) {
         val (nps, notAtAll, extremely) = createRefs()
         Row(
-            modifier = Modifier
-                .constrainAs(nps) {
-                    start.linkTo(parent.start, 20.dp)
-                    end.linkTo(parent.end, 20.dp)
-                    top.linkTo(parent.top)
-                },
-            horizontalArrangement = Arrangement.Center
+            modifier = Modifier.constrainAs(nps) {
+                start.linkTo(parent.start, 20.dp)
+                end.linkTo(parent.end, 20.dp)
+                top.linkTo(parent.top)
+            }, horizontalArrangement = Arrangement.Center
         ) {
             answers.forEachIndexed { index, surveyAnswer ->
-                val eachItemWidth = LocalDensity.current.run {
-                    (LocalConfiguration.current.screenWidthDp.dp - 40.dp) / answers.size
-                }
-                TextButton(
-                    onClick = {
-                        selectedIndex = index
-                        onChooseAnswer(
-                            answers.mapIndexed { index, surveyAnswer ->
+                if (index < MAX_NUMBER_OF_NPS_QUESTION) {
```

Comment:
We could update the `answers.forEachIndexed` call instead of adding this `if` check by using `.take()` 😉 
So the call above would be 
`answers.take(MAX_NUMBER_OF_NPS_QUESTION).forEachIndexed { index, surveyAnswer ->` 😄 

The rest LGTM 👍🏻